File: game-store-react/src/entities/Game.ts
export interface GamePrice {
  amount: number;
  currency: string;
}

export interface Game {
  id: number;
  name: string;
  genreName: string;
  genreId: number;
  price: GamePrice;
  releaseDate: string;
}

export type GameFormData = Omit<Game, 'id'>;

File: GameStore.Api/src/Domain/Entities/Game.cs
using MyGameStore.Domain.ValueObjects;

namespace GameStore.Api.Models;

public class Game
{
    public int Id { get; set; }

    public required GameTitle Title { get; set; }

    public Genre? Genre { get; set; }

    public int GenreId { get; set; }

    public required GamePrice Price { get; set; }

    public required ReleaseDate ReleaseDate { get; set; }

    public bool IsAffordable(decimal userBalance) => userBalance >= Price.Amount;

    public bool IsDiscounted(decimal originalPrice) => Price.Amount < originalPrice;
}
